api/api-rules/violation_exceptions.list

```diff
@@ -58,6 +58,10 @@ API rule violation: list_type_missing,k8s.io/api/authorization/v1beta1,ResourceR
 API rule violation: list_type_missing,k8s.io/api/authorization/v1beta1,SubjectAccessReviewSpec,Groups
 API rule violation: list_type_missing,k8s.io/api/authorization/v1beta1,SubjectRulesReviewStatus,NonResourceRules
 API rule violation: list_type_missing,k8s.io/api/authorization/v1beta1,SubjectRulesReviewStatus,ResourceRules
+API rule violation: list_type_missing,k8s.io/api/autoscaling/v2,HPAScalingRules,Policies
+API rule violation: list_type_missing,k8s.io/api/autoscaling/v2,HorizontalPodAutoscalerSpec,Metrics
+API rule violation: list_type_missing,k8s.io/api/autoscaling/v2,HorizontalPodAutoscalerStatus,Conditions
+API rule violation: list_type_missing,k8s.io/api/autoscaling/v2,HorizontalPodAutoscalerStatus,CurrentMetrics
```

Comment:
@apelisse @jpbetz can you recommend a list type here? would like to see us resolve these rather than propagate them into a GA API version

Reply:
Thanks! Sorry, it is first time to me that I try to promote an API version.  Shall we have a commit  without any files changed (only copy v2beta2 to v2)?
My workflow is: 1.copying v2beta2 to v2,  2. changing the files which need to be changed , 3. testing on the local,  4. moving the types.go  out of "kubernetes" dir, 5. adding  a commit for rest, 6. moving the types.go into  "kubernetes"  dir , 7. adding  a commit for types.go. 

conversions and defaults have be changed. I have just renamed the package name, fields name which should be to renamed  as #102362 said. 

As if testing will failed if we are not adding "API rule violation: ......" into violation_exceptions.list file.

Am I must redo this PR?

Reply:
> Shall we have a commit without any files changed (only copy v2beta2 to v2)?

yeah, when promoting, the following structure is helpful:
1. a commit that just copies the `pkg/apis/autoscaling/v2beta2` and `staging/src/k8s.io/api/autoscaling/v2beta2` packages to v2
2. a commit that renames v2beta2 to v2 in the new files
3. a commit that actually makes any changes introduced in v2

That makes it easy to verify mechanical changes and see the deltas.

> Am I must redo this PR?

If you can set those commits up in this PR, I would find that helpful. Not sure about the other reviewers.

Reply:
These should be resolved and dropped

Reply:
Ok. I have deleted the lines which be added by this PR from api/api-rules/violation_exceptions.list

Reply:
this is still outstanding... a `// +listType=...` comment needs to be added to these fields so these exceptions are no longer required

Reply:
1. I am remember deleting the lines from the exception file. I was try to redo this. And I found that the lines  are  added into the file when I run the following command as api/api-rules/README.md said:
make generated_files UPDATE_API_KNOWN_VIOLATIONS=true
2. As if pull-kubernetes-verify-govet-levee test fails when I try to remove the lines from api/api-rules/violation_exceptions.list file. Error message is: ERROR: KUBE API rule check failed. Reported violations differ from known violations. Please read api/api-rules/README.md to resolve the failure in api/api-rules/violation_exceptions.list.  Ref: [https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/102534/pull-kubernetes-verify-govet-levee/1440863547327254528](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/102534/pull-kubernetes-verify-govet-levee/1440863547327254528)
3.  I was try to fix test fails by rebase from upstream/master. But I can not fix the fails. Could you give me a hand? Thanks.

Reply:
> 1. I found that the lines  are  added into the file when I run the following command as api/api-rules/README.md said:
>    make generated_files UPDATE_API_KNOWN_VIOLATIONS=true

That's because the issues indicated here need to be resolved by defining a list type for the fields. See https://github.com/kubernetes/kubernetes/pull/102534#discussion_r714077788. Work with @apelisse to select the appropriate list type for these fields.